mobile/lib/src/providers/pipelines_provider.dart

```diff
@@ -1,96 +1,90 @@
-    ], sortingMethod: PipelineCollectionSort.last, sortingSplitDisabled: true);
+    _pipelineCollection = PipelineCollection(
+        pipelines: [],
+        sortingMethod: PipelineCollectionSort.last,
+        sortingSplitDisabled: true);
+    fetchPipelines();
   }
 
-  /// Adds a pipeline in the Provider
-  addPipelineInProvider(Pipeline newPipeline) {
-    pipelineCollection.pipelines.add(newPipeline);
-    sortPipelines();
-    notifyListeners();
+  /// Fetches the pipelines from API and put them in the collection
+  Future<void> fetchPipelines() {
+    return GetIt.I<AerisAPI>().getPipelines().then((pipelines) {
+      _pipelineCollection.pipelines = pipelines;
+      sortPipelines();
+      initialized = true;
+    });
   }
 
-  /// Sets a new list of pipelines into the Provider
-  setPipelineProvider(List<Pipeline> newPipelines) {
-    pipelineCollection.pipelines = [];
-    pipelineCollection.pipelines = newPipelines;
+  /// Adds a pipeline in the Provider
+  addPipeline(Pipeline newPipeline) {
+    initialized = true;
+    _pipelineCollection.pipelines.add(newPipeline);
+    GetIt.I<AerisAPI>().createPipeline(newPipeline);
     sortPipelines();
+    notifyListeners();
   }
 
-  /// Sort pipelines inside the Provider
+  /// Sort pipelines inside the Provider, and notify listeners
   sortPipelines() {
-    pipelineCollection.sort();
+    _pipelineCollection.sort();
     notifyListeners();
   }
 
-  /// Removes a specific pipeline from the Provider
+  /// Removes a specific pipeline from the Provider, and notify listeners
   removePipeline(Pipeline pipeline) {
-    pipelineCollection.pipelines.remove(pipeline);
+    _pipelineCollection.pipelines.remove(pipeline);
+    GetIt.I<AerisAPI>().removePipeline(pipeline);
     notifyListeners();
   }
 
-  /// Removes every pipeline from the Provider
-  clearProvider() {
-    pipelineCollection.pipelines.clear();
+  /// Removes piplines matching predicates, and notify listeners
+  removePipelinesWhere(bool Function(Pipeline) predicate) {
+    List<Pipeline> toRemove =
+        _pipelineCollection.pipelines.where(predicate).toList();
+    for (Pipeline pipeline in toRemove) {
+      GetIt.I<AerisAPI>().removePipeline(pipeline);
+      _pipelineCollection.pipelines.remove(pipeline);
+    }
     notifyListeners();
   }
+
+  /// returns the number of piepliens currently in the collection
+  int getPipelineCount() {
+    return _pipelineCollection.pipelines.length;
+  }
+
+  /// returns the pipeline a t the given index in the collection.
+  /// Warning: if the pipeline is updated, no call will be made to api
+  Pipeline getPipelineAt(int index) {
+    return _pipelineCollection.pipelines[index];
+  }
+
+  /// Sets sorting method for pipelines, does not execute sorting
+  void setSortingMethod(PipelineCollectionSort sortingMethod) {
+    _pipelineCollection.sortingMethod = sortingMethod;
+  }
+
+  /// Sets sorting method for pipelines, does not execute sorting
+  void setDisabledSplit(bool split) {
```

Comment:
Use properties to do that, we don't want raw getter/setter. See [this](https://stackoverflow.com/questions/27683924/how-do-getters-and-setters-change-properties-in-dart) for more references.